clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1025,23 +1025,6 @@ class KernelObjVisitor {
     VisitRecordFields(Owner, Handlers...);
   }
 
-  // FIXME: Can this be refactored/handled some other way?
-  template <typename ParentTy, typename... HandlerTys>
-  void visitStreamRecord(const CXXRecordDecl *Owner, ParentTy &Parent,
-                         CXXRecordDecl *Wrapper, QualType RecordTy,
-                         HandlerTys &... Handlers) {
-    (void)std::initializer_list<int>{
-        (Handlers.enterStream(Owner, Parent, RecordTy), 0)...};
```

Comment:
I think you can remove the corresponding enterStream/leaveStream definitions everywhere now. With this change I don't think you're using it anymore anywhere. Stream class is no longer decomposed i.e. it is handled as a whole without handling it's individual accessors.